src/QMCTools/ppconvert/src/CMakeLists.txt

```diff
@@ -8,6 +8,8 @@ target_sources(
          "${CMAKE_CURRENT_SOURCE_DIR}/XMLWriterClass2.cc" "${CMAKE_CURRENT_SOURCE_DIR}/NLPPClass.cc"
          "${CMAKE_CURRENT_SOURCE_DIR}/ParserClass.cc")
 
+target_compile_options(ppconvert PUBLIC -fno-fast-math)  # ppconvert generates and checks for NaN in normal use; needs IEEE compliance
```

Comment:
It can potentially choke non GCC/Clang compiler I guess. Does this work with all C++ compilers?

Reply:
I guess it works in all compilers that support `-ffast-math`.

Reply:
So it can be a hassle and I don't think we should take this route.

Reply:
Do you think `-fno-fast-math` is not supported in all the compilers that matter?

Reply:
```
$ nvc++ -fno-fast-math main.cpp 
nvc++-Error-Unknown switch: -fno-fast-math
```

Reply:
good catch, is there a CI test for nvc++ currently?

Reply:
We do have intel classic. As I suggested, write a test in cmake checking `std::isnan` and turn off ppconvert if the test fails.

Reply:
The problem is that these kind of checks are fragile, depend on running the code (never is going to work for crosscompilation), and  basically, we are saying that `ppconvert` is going to be compiled or not compiled "randomly" depending on the compiler and flags that leak from the global settings. For that matter we can only compile it with GCC and clang, at least we will know when it is compiled and when it is not.

Reply:
ppconvert should be a host tool. With crosscompilation, the results of test failing is not producing a ppconvert, no harm.